### ISSUE TITLE
improve cache decorator

### ISSUE THREAD

**Issue by josef-pkt** [statsmodels/statsmodels#886]
Is it possible to improve introspection and documentation from docstring for cached properties?

see #885 for some discussion

**Comment by jbrockmendel** [comment]
The pandas version should be used.  More performing, better docstring behavior, and reduce code redundancy.

**Comment by josef-pkt** [comment]
I don't remember what the issue was, and don't want to get into the details again.

"If it ain't broken, don't fix it."
Because we make very heavy use of the cache decorator in the results, changes might be dangerous.
I have not much of an idea how good our test coverage would be for screwing up the cache decorator, e.g. by getting a new backwards incompatible version, e.g. test isolation might not provide checking for messed up cache behavior.
There would need to be a large advantage to non-trivial changes to compensate for the risk.

**Comment by jbrockmendel** [comment]
> I don't remember what the issue was, and don't want to get into the details again.
> "If it ain't broken, don't fix it."

Seems like an argument for closing this issue?

> I have not much of an idea how good our test coverage would be for screwing up the cache decorator, e.g. by getting a new backwards incompatible version, e.g. test isolation might not provide checking for messed up cache behavior.

This mirrors my attitude about the test suite in general.  

> There would need to be a large advantage to non-trivial changes to compensate for the risk.

A few judgement-free thoughts:

- In the short term risk aversion makes sense.  Longer term, less code means less maintenance burden.

- Willingness to sort through non-stats parts of statsmodels for simplification/consistency improvements is an area where I have a strong comparative advantage.  e.g. I've determined that we make zero usage of `ResettableCache` that cannot be replaced by a dict.

- You opened an issue a little while ago about a low-memory variant of `cache_readonly`, which I think is a good enough idea to constitute "large advantage".  Luckily I've already implemented it, and it also solves the `_data_attributes`-slipping-through-the-cracks problem.  This can be discussed in its extant PR.
